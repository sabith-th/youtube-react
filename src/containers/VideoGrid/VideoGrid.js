import React from "react";
import { Divider } from "semantic-ui-react";
import { VideoPreview } from "../VideoPreview/VideoPreview";
import "./VideoGrid.scss";
import { VideoGridHeader } from "./VideoGridHeader/VideoGridHeader";

export const VideoGrid = props => {
  const divider = props.hideDivider ? null : <Divider />;
  return (
    <React.Fragment>
      <VideoGridHeader title={props.title} />
      <div className="video-grid">
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      {divider}
    </React.Fragment>
  );
};
