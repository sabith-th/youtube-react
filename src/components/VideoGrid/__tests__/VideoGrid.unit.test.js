import { shallow } from "enzyme";
import React from "react";
import { VideoGrid } from "../VideoGrid";

describe("Video Grid", () => {
  test("renders without props", () => {
    const wrapper = shallow(<VideoGrid />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with title prop", () => {
    const wrapper = shallow(<VideoGrid title="Trending" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders without divider", () => {
    const wrapper = shallow(<VideoGrid hideDivider={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
