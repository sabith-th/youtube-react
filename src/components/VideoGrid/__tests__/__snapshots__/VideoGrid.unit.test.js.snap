// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Grid renders with title prop 1`] = `<div />`;

exports[`Video Grid renders without divider 1`] = `<div />`;

exports[`Video Grid renders without props 1`] = `<div />`;
